x-pack/plugins/observability_solution/slo/public/components/slo/delete_confirmation_modal/slo_delete_confirmation_modal.tsx

```diff
@@ -5,50 +5,158 @@
  * 2.0.
  */
 
-import { EuiConfirmModal } from '@elastic/eui';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSwitch,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
 import { ALL_VALUE, SLODefinitionResponse, SLOWithSummaryResponse } from '@kbn/slo-schema';
-import React from 'react';
-import { getGroupKeysProse } from '../../../utils/slo/groupings';
+import React, { useState } from 'react';
+import { useDeleteSlo } from '../../../hooks/use_delete_slo';
+import { useDeleteSloInstance } from '../../../hooks/use_delete_slo_instance';
 
-export interface SloDeleteConfirmationModalProps {
+export interface Props {
   slo: SLOWithSummaryResponse | SLODefinitionResponse;
   onCancel: () => void;
-  onConfirm: () => void;
+  onSuccess: () => void;
 }
 
-export function SloDeleteConfirmationModal({
-  slo,
-  onCancel,
-  onConfirm,
-}: SloDeleteConfirmationModalProps) {
+export function SloDeleteModal({ slo, onCancel, onSuccess }: Props) {
   const { name, groupBy } = slo;
+  const instanceId =
+    'instanceId' in slo && slo.instanceId !== ALL_VALUE ? slo.instanceId : undefined;
```

Comment:
this condition is probably used gazillion times in code :) 

i do kinda annoy me that we stored * for all values, i wonder if keeping it undefined would have helped us

Reply:
so technically, we are always backfilling instanceId when reading from the API. Sooooo, we could update the type to reflect that :)

Reply:
there is even an issue for it https://github.com/elastic/kibana/issues/180590